src/main/java/org/folio/circulation/domain/ChangeDueDateRepository.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.circulation.domain;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ServerErrorFailure;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ChangeDueDateRepository {
+
+  private static final Logger log = LoggerFactory
+    .getLogger(MethodHandles.lookup().lookupClass());
+
+  private final LoanRepository loanRepository;
+
+  public ChangeDueDateRepository(Clients clients) {
+    loanRepository = new LoanRepository(clients);
+  }
+
+  public CompletableFuture<Result<Loan>> changeDueDate(String loanId,
+    DateTime dueDate) {
+
+    log.info("Changing due date for {} to {}", loanId, dueDate);
+    try {
+      return loanRepository.getById(loanId)
```

Comment:
Why are all of the loans being fetched individually? One of the advantages of batching might be to fetch all of the (or a page of) loans together, rather than fetching each one in turn.